test/tests/smoke_test.rs

```diff
@@ -50,4 +99,53 @@ fn smoke_test() {
 | |__| (_| | | | |_) | |_| | | (_| | |  _ < | |
  \____\__,_|_|_| .__/ \__|_|  \__,_| |_| \_\|_|"#,
     );
+
+    const TEST_ONLY_GET_LDEV_CERT: u32 = 0x4345524c; // "CERL"
+    const TEST_ONLY_GET_FMC_ALIAS_CERT: u32 = 0x43455246; // "CERF"
+
+    let ldev_cert_der = hw
+        .mailbox_execute(TEST_ONLY_GET_LDEV_CERT, &[])
+        .unwrap()
+        .unwrap();
+    let ldev_cert = openssl::x509::X509::from_der(&ldev_cert_der).unwrap();
+    let ldev_cert_txt = String::from_utf8(ldev_cert.to_text().unwrap()).unwrap();
+
+    // To update the ldev cert testdata:
+    // std::fs::write("tests/smoke_testdata/ldevid_cert.txt", &ldev_cert_txt).unwrap();
+    // std::fs::write("tests/smoke_testdata/ldevid_cert.der", &ldev_cert_der).unwrap();
+
+    assert_eq!(
+        ldev_cert_txt.as_str(),
+        include_str!("smoke_testdata/ldevid_cert.txt")
```

Comment:
What's the rationale for comparing against both the der and text?

Reply:
I'm not sure whether openssl includes every bit of the cert in the text, so this way we have both a human-readable diff in any PR that changes this logic, plus the actual bits for exhaustive assertion.

Reply:
SGTM